File: backend/src/main/java/com/docloader/service/DocumentService.java
package com.docloader.service;

import com.docloader.model.Document;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface DocumentService {
    
    /**
     * Get a document by its ID
     * @param id the document ID
     * @return the document if found
     */
    Optional<Document> getDocumentById(UUID id);
    
    /**
     * Get all documents for a job
     * @param jobId the job ID
     * @return list of documents
     */
    List<Document> getDocumentsByJobId(UUID jobId);
    
    /**
     * Save a document
     * @param document the document to save
     * @return the saved document
     */
    Document saveDocument(Document document);
    
    /**
     * Delete a document
     * @param id the document ID
     */
    void deleteDocument(UUID id);
    
    /**
     * Get the count of documents for a user
     * @param userId the user ID
     * @return the count of documents
     */
    int getDocumentCountByUserId(UUID userId);
} 

File: frontend/src/services/documentService.js
import api from './axiosConfig';

const documentService = {
  // Upload documents
  uploadDocuments: async (files) => {
    try {
      const formData = new FormData();
      files.forEach(file => {
        formData.append('files', file);
      });
      
      const response = await api.post('/documents/upload', formData, {
        headers: {
          'Content-Type': 'multipart/form-data'
        }
      });
      
      return response.data;
    } catch (error) {
      console.error('Error uploading documents:', error);
      throw error;
    }
  },
  
  // Get all documents for current user
  getUserDocuments: async () => {
    try {
      const response = await api.get('/documents');
      return response.data;
    } catch (error) {
      console.error('Error fetching user documents:', error);
      throw error;
    }
  },
  
  // Get document by ID
  getDocumentById: async (documentId) => {
    try {
      const response = await api.get(`/documents/${documentId}`);
      return response.data;
    } catch (error) {
      console.error(`Error fetching document ${documentId}:`, error);
      throw error;
    }
  },
  
  // Get documents for a job
  getDocumentsByJobId: async (jobId) => {
    try {
      const response = await api.get(`/documents/job/${jobId}`);
      return response.data;
    } catch (error) {
      console.error(`Error fetching documents for job ${jobId}:`, error);
      throw error;
    }
  },
  
  // Delete a document
  deleteDocument: async (documentId) => {
    try {
      await api.delete(`/documents/${documentId}`);
      return true;
    } catch (error) {
      console.error(`Error deleting document ${documentId}:`, error);
      throw error;
    }
  }
};

export default documentService; 